src/indexBy.ts

```diff
@@ -52,15 +67,42 @@ const _indexBy =
     }, {});
   };
 
+export function indexByStrict<K extends PropertyKey, T>(
+  array: ReadonlyArray<T>,
+  fn: (item: T) => K
+): Partial<Record<K, T>>;
+
+export function indexByStrict<K extends PropertyKey, T>(
+  fn: (item: T) => K
+): (array: ReadonlyArray<T>) => Partial<Record<K, T>>;
+
+export function indexByStrict() {
+  return purry(_indexByStrict(false), arguments);
+}
```

Comment:
You are exporting indexByStrict which would make it accessible directly (and not via the namespace accessor).
```suggestion
function indexByStrict<K extends PropertyKey, T>(
  array: ReadonlyArray<T>,
  fn: (item: T) => K
): Partial<Record<K, T>>;

function indexByStrict<K extends PropertyKey, T>(
  fn: (item: T) => K
): (array: ReadonlyArray<T>) => Partial<Record<K, T>>;

function indexByStrict() {
  return purry(_indexByStrict(false), arguments);
}
```